lib/puppet/configurer.rb

```diff
@@ -126,6 +126,82 @@ def convert_catalog(result, duration, facts, options = {})
     catalog
   end
 
+  def check_fact_name_length(name)
+    unless Puppet[:max_factnamelength] == -1 
+      if Puppet[:max_factnamelength] == 0
+        Puppet[:max_factnamelength] = 100
+        if name.length > Puppet[:max_factnamelength]
+          Puppet.warning _("Fact %{name} exceedes the lenght limit: %{limit}") % { name: name, limit: Puppet[:max_factnamelength] }
+        end
+      else
+        if name.length > Puppet[:max_factnamelength]
+          Puppet.warning _("Fact %{name} exceedes the lenght limit: %{limit}") % { name: name, limit: Puppet[:max_factnamelength] }
```

Comment:
This and other messages should also specify the current value, something like: `Fact %{name} has the length %{length} which exceedes the length limit ...`